drivers/i2c/i2c_emul.c

```diff
@@ -89,6 +90,41 @@ static int i2c_emul_send_to_target(const struct device *dev, struct i2c_msg *msg
 	struct i2c_emul_data *data = dev->data;
 	const struct i2c_target_callbacks *callbacks = data->target_cfg->callbacks;
 
+#ifdef CONFIG_I2C_TARGET_BUFFER_MODE
+	const struct i2c_emul_config *config = dev->config;
+
+	if (!config->target_pio_mode) {
```

Comment:
This should be `target_buffer_mode`.  Programmed I/O usually means write/read one byte at time.

Reply:
I modeled this after the `i2c_ite_enchance` driver which uses the DT property `target-pio-mode` in order to default to the buffered mode.

Reply:
The `target-pio-mode` option is independent of the buffer mode API.  The buffer mode API is a property of the I2C target driver (such as the [eeprom_target](https://github.com/zephyrproject-rtos/zephyr/blob/af314643a32db3452dd77f20dee2807197b179cf/drivers/i2c/target/eeprom_target.c#L227)) and this really isn't a property of the I2C controller driver.

I do see that the `i2c_ite_enhance` driver does reference the `buf_write_received` and `buf_read_received` routines, but this is only done for DMA mode (`target-pio-mode = false`).